cdi/src/test/java/io/smallrye/config/inject/ValidateInjectionTest.java

```diff
@@ -304,8 +562,67 @@ void fail() {
         @ApplicationScoped
         static class MissingPropertyExpressionBean {
             @Inject
-            @ConfigProperty(name = "bad.property.expression.prop")
+            @ConfigProperty(name = "bad.property.expression.prop") // Exists but contains ${missing.prop} which doesn't 
             String missing;
         }
     }
+
+    @ExtendWith(WeldJunit5Extension.class)
+    static class ManyInjectionExceptionsTest extends InjectionTest {
+        @WeldSetup
+        WeldInitiator weld = WeldInitiator.from(ConfigExtension.class, ManyInjectionExceptionsBean.class)
+                .addBeans()
+                .activate(ApplicationScoped.class)
+                .inject(this)
+                .build();
+
+        @Inject
+        ManyInjectionExceptionsBean bean;
+
+        @Test
+        void fail() {
+            Assertions.fail();
+        }
+
+        @ApplicationScoped
+        static class ManyInjectionExceptionsBean {
+            @Inject
+            @ConfigProperty(name = "missing.property")
+            String missingProp;
+
+            @Inject
+            @ConfigProperty(name = "empty.property")
+            String emptyProp;
+
+            @Inject
+            @ConfigProperty(name = "bad.property")
+            String[] badProp;
+        }
+    }
+
+    @ExtendWith(WeldJunit5Extension.class)
+    static class UnqualifiedConfigPropertiesInjectionTest extends InjectionTest {
+        @WeldSetup
+        WeldInitiator weld = WeldInitiator.from(ConfigExtension.class, ServerDetailsBean.class)
+                .addBeans()
+                .activate(ApplicationScoped.class)
+                .inject(this)
+                .build();
+
+        @Inject
+        // Unqualified due to @ConfigProperties missing here
+        ServerDetailsBean server;
```

Comment:
I'm not sure this test is actually testing config-injection functionality.

Isn't it just checking how Weld handles a bad Injection?

Reply:
The current output is just:
```
org.jboss.weld.exceptions.IllegalArgumentException: WELD-001408: Unsatisfied dependencies ...
...
Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-001408: Unsatisfied dependencies ...
...
```